objects.go

```diff
@@ -63,11 +97,130 @@ type Network struct {
+	MgmtPhysicalSetting  map[string]bool        `json:"mgmt_physical_setting,omitempty"`
+	PaidNios             bool                   `json:"paid_nios,omitempty"`
+	PhysicalOid          string                 `json:"physical_oid,omitempty"`
+	ServiceStatus        []ServiceStatus        `json:"service_status,omitempty"`
+	V6MgmtNetworkSetting map[string]interface{} `json:"v6_mgmt_network_setting,omitempty"`
+}
+
+type Member struct {
+	IBBase                   `json:"-"`
+	Ref                      string `json:"_ref,omitempty"`
+	HostName                 string `json:"host_name,omitempty"`
+	ConfigAddrType           string `json:"config_addr_type,omitempty"`
+	PLATFORM                 string `json:"platform,omitempty"`
+	ServiceTypeConfiguration string `json:"service_type_configuration,omitempty"`
+}
+
+// License represents license wapi object
+type License struct {
+	IBBase           `json:"-"`
+	Ref              string `json:"_ref,omitempty"`
+	ExpirationStatus string `json:"expiration_status,omitempty"`
+	ExpiryDate       int    `json:"expiry_date,omitempty"`
+	HwID             string `json:"hwid,omitempty"`
+	Key              string `json:"key,omitempty"`
+	Kind             string `json:"kind,omitempty"`
+	Limit            string `json:"limit,omitempty"`
+	LimitContext     string `json:"limit_context,omitempty"`
+	Licensetype      string `json:"type,omitempty"`
+}
+
+type MemberResult struct {
+	Ref      string     `json:"_ref,omitempty"`
+	HostName string     `json:"host_name,omitempty"`
+	Nodeinfo []NodeInfo `json:"node_info,omitempty"`
+	TimeZone string     `json:"time_zone"`
+}
+
+// CapacityReport represents capacityreport object
+type CapacityReport struct {
+	IBBase `json:"-"`
+	Ref    string `json:"_ref,omitempty"`
+
+	Name         string                   `json:"name,omitempty"`
+	HardwareType string                   `json:"hardware_type,omitempty"`
+	MaxCapacity  int                      `json:"max_capacity,omitempty"`
+	ObjectCount  []map[string]interface{} `json:"object_counts,omitempty"`
+	PercentUsed  int                      `json:"percent_used,omitempty"`
+	Role         string                   `json:"role,omitempty"`
+	TotalObjects int                      `json:"total_objects,omitempty"`
+}
+type NTPSetting struct {
+	enable_ntp bool                   `json:"enable_ntp,omitempty"`
+	NTPAcl     map[string]interface{} `json:"ntp_acl,omitempty"`
+	NTPKeys    []string               `json:"ntp_keys,omitempty"`
+	NTPKod     bool                   `json:"ntp_kod,omitempty"`
+	NTPServers []string               `json:"ntp_servers,omitempty"`
+}
+
+type Grid struct {
+	IBBase `json:"-"`
+	Ref    string `json:"_ref,omitempty"`
+}
+
+type GridResult struct {
+	Ref        string     `json:"_ref,omitempty"`
+	Name       string     `json:"name,omitempty"`
+	NTPSetting NTPSetting `json:"ntp_setting"`
+}
+
+func NewGrid(grid Grid, returnFields []string) *Grid {
```

Comment:
Is it really necessary for `returnFields` to be passed in?

I can see that it might be useful, for efficient reason, to allow user to specify the return fields if the result structure has many large fields that user often don't need. Is that the case?

Still, even if this is the case, the method should set a default set of return fields.

OK... looking at the object manager code, it looks the returnFields are used in the GetXyZ() methods and are always passed the same fixed list of returnFields. I am therefore included to say remove returnFields parameters in this and other NewXyZ() methods and set the defaults inside the methods.

Reply:
i felt  NewXyz should take params, if not, if someone wants specify only subset of returnFields then he will have to modify NewXyz, we better avoid that scenario.

Object_manager can grow with  various getXyz, getXwy, etc.

Reply:
I agree with you that there may be future use cases that require the ability to modify return fields.

However, that still doesn't justify forcing user to pass in returnFields in NewXyZ(), because returnFields is only relevant if the user wants to do a get/search.

Please consider, instead, adding a method SetReturnFields() for setting returnFields. The NewXyZ() should still define a default list of returnFields that include all the attributes defined for the object.

Note that, realistically, this only allow user to remove fields that he doesn't need returned, since you cannot add any return field that is not already an attribute of the object.

Reply:
ok, so you mean to say expand the `IBObject` interface to have SetReturnFields method ? that will enable modifying the _return_fields ?

Reply:
Yes.